src/workerd/api/node/zlib-util.h

```diff
@@ -173,42 +178,72 @@ class ZlibContext {
 
 using CompressionStreamErrorHandler = jsg::Function<void(int, kj::StringPtr, kj::StringPtr)>;
 
-template <typename CompressionContext>
-class CompressionStream {
+class BrotliContext {
 public:
-  CompressionStream() = default;
-  ~CompressionStream() noexcept(false);
-  KJ_DISALLOW_COPY_AND_MOVE(CompressionStream);
-
-  void close();
-  bool checkError(jsg::Lock& js);
-  void emitError(jsg::Lock& js, const CompressionError& error);
-  template <bool async>
-  void writeStream(jsg::Lock& js,
-      int flush,
-      kj::ArrayPtr<kj::byte> input,
+  explicit BrotliContext(ZlibMode _mode): mode(_mode) {}
+  KJ_DISALLOW_COPY(BrotliContext);
+  void setBuffers(kj::ArrayPtr<kj::byte> input,
       uint32_t inputLength,
       kj::ArrayPtr<kj::byte> output,
       uint32_t outputLength);
-  void setErrorHandler(CompressionStreamErrorHandler handler) {
-    errorHandler = kj::mv(handler);
-  };
-  void initializeStream(jsg::BufferSource _write_result, jsg::Function<void()> writeCallback);
-  void updateWriteResult();
+  void setFlush(int flush);
+  void getAfterWriteResult(uint32_t* availIn, uint32_t* availOut) const;
+  void setMode(ZlibMode _mode) {
+    mode = _mode;
+  }
 
 protected:
-  CompressionContext context;
+  ZlibMode mode;
+  const uint8_t* nextIn = nullptr;
+  uint8_t* nextOut = nullptr;
+  size_t availIn = 0;
+  size_t availOut = 0;
+  BrotliEncoderOperation flush = BROTLI_OPERATION_PROCESS;
+
+  // TODO(addaleax): These should not need to be stored here.
```

Comment:
Probably a good idea to add some extra comments at the top of this file pointing out that a lot of this was adapted from Node.js and therefore follows a lot of Node.js style as opposed to kj style. Some folks coming into this code later on might not have any idea who or what an "addaleax" is ;-)

Reply:
Agreed. Just added.